### PR TITLE
Controller mounting tweaks

### PR DESCRIPTION
**Pull request by CarsonF**
Updated `Application::mount` to forward to our mount event. This ensures low priority (less than 0) controllers are last.

Cleaned up some of the Controller SP as well.

### CODE REVIEW

**Comment by CarsonF** [comment]
Can I get some thoughts on this? I would like App::mount to work well with our new mounting event, but I don't want to screw up any preconceived ideas. Maybe the connect method should be called immediately, then passed to listener to mount it to the collection?

**Comment by GawainLynch** [comment]
Can you ping me about this tomorrow in IRC.

File: src/Events/MountEvent.php [hunk 1]
```diff
@@ -67,10 +68,47 @@ public function finish()
...
+    /**
+     * Verifies collection is correct type and calls connect on providers.
+     *
+     * Note: This is the same code as {@see Silex\Application::mount}
+     *
+     * @param ControllerProviderInterface|ControllerCollection $collection
+     *
+     * @return ControllerCollection
+     *
+     * @throws LogicException If controllers is not an instance of ControllerProviderInterface or ControllerCollection
```
Comment (GawainLynch) [inline src/Events/MountEvent.php]: @param
@throws
@return

[@see](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#examples-18)

**Comment by GawainLynch** [comment]
A rebase would be awesome if you could… 

Either way, from how I am reading it I think I prefer the current approach… K.I.S.S. at least for now. :+1:

**Comment by CarsonF** [comment]
Ok I'm happy where this ended up :shipit:

**Comment by GawainLynch** [comment]
The OCD is strong in this one!

**Pull request merged**